### PR TITLE
configure project version to user agent

### PR DESCRIPTION
**Pull request by sophiaso**

### CODE REVIEW

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/utils/PropertyLoader.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private static final String PROJECT_PROPERTY_FILE = "/project.properties";
+
+    public static String getProjectVersion() {
+        String version = "unknown";
+        try (InputStream inputStream = PropertyLoader.class.getResourceAsStream(PROJECT_PROPERTY_FILE)) {
+            if (inputStream != null) {
+                final Properties properties = new Properties();
+                properties.load(inputStream);
+
+                version = properties.getProperty("project.version");
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/utils/PropertyLoader.java]: not sure if `hasText` check is need here but just corner case.

File: pom.xml [hunk 2]
```diff
@@ -218,6 +218,12 @@
                 </executions>
             </plugin>
         </plugins>
+        <resources>
```
Comment (warrenzhu25) [inline pom.xml]: Maybe you should follow other modules about how to get a project version.

Comment (sophiaso) [reply pom.xml]: The side effect of `Class#getPackage().getImplementationVersion()` is that, the runtime implementation will always be null.  Or we should consider it as not a problem?

Comment (sophiaso) [reply pom.xml]: Updated to use implementation version directly, ignore not packaged phase null version value case.

**Pull request merged**